csrc/type_shim.h

```diff
@@ -3,26 +3,26 @@
 // Forward/backward compatiblity hack around
 // https://github.com/pytorch/pytorch/commit/3aeb78079bcd68282fe9117088e138b77318e288
 // pending more future-proof guidance from upstream.
-struct TypeShim
-{
-  const at::Type& payload;
-  TypeShim(const at::Type& type) : payload(type) {}
-  // Enable trivial conversion to a const at::Type& for pre-3aeb78
-  operator const at::Type&(){ return payload; };
-  // Enable dispatch switch statements to take *this directly for  post-3aeb78
-  operator at::ScalarType(){ return payload.scalarType(); };
-};
+// struct TypeShim
+// {
+//   const at::Type& payload;
+//   TypeShim(const at::Type& type) : payload(type) {}
+//   // Enable trivial conversion to a const at::Type& for pre-3aeb78
+//   operator const at::Type&(){ return payload; };
+//   // Enable dispatch switch statements to take *this directly for  post-3aeb78
+//   //operator at::ScalarType(){ return payload.; };
+// };
 
 #define DISPATCH_FLOAT_AND_HALF(TYPE, LEVEL, NAME, ...) \
   switch(TYPE) \
   { \
-    case at::ScalarType::Float: \
+    case at::kFloat: \
```

Comment:
I feel more comfortable with the existing `at::ScalarType::Float` syntax because that's what the upstream dispatch macros currently use https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h

Reply:
I've changed it back to the `at::ScalarType::Type` usage for most of the code. I only left the usage of `at::kType` for creating new tensors (in case `tensor.scalar_type()` wasn't used), as it seems to be the recommended way ([link](https://github.com/pytorch/pytorch/blob/c42f3f9055b1ead5253fb99980bfaf46ecafcb0f/c10/core/TensorOptions.h#L49)).

Would this be alright or should I revert all usages of `at::kType`?

Reply:
I feel most comfortable mirroring what upstream does, so this seems fine